executor_manager/services/task_queue_service.py

```
@@ -0,0 +1,253 @@
+            return []
+
+        try:
+            # LRANGE with negative indices gets from right (oldest)
+            task_jsons = self.redis_client.lrange(self.queue_key, -count, -1)
+            tasks = []
+            for task_json in task_jsons:
+                try:
+                    tasks.append(json.loads(task_json))
+                except json.JSONDecodeError:
+                    continue
+            return tasks
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)